tests/backup/backup_kubevirt_test.go

```diff
@@ -1760,3 +1765,424 @@ var _ = Describe("{KubevirtInPlaceRestoreWithReplaceAndRetain}", func() {
+				if err != nil {
+					log.Errorf("TransformAppContextWithMappings: %v", err)
+					continue
+				}
+				newRestoredAppContexts = append(newRestoredAppContexts, c)
+			}
+			vms, err := GetAllVMsFromScheduledContexts(newRestoredAppContexts)
+			log.FailOnError(err, "Failed to get VMs from scheduled contexts")
+			for _, v := range vms {
+				t := func() (interface{}, bool, error) {
+					diskCountOutput, err := GetNumberOfDisksInVM(v)
+					if err != nil {
+						return nil, false, fmt.Errorf("failed to get number of disks in VM [%s] in namespace [%s]", v.Name, v.Namespace)
+					}
+					// Total disks will be numberOfVolumes plus the container disk
+					if diskCountOutput != diskCount+numberOfAdditionalDisks {
+						return nil, true, fmt.Errorf("expected number of disks in VM [%s] in namespace [%s] is [%d] but got [%d]", v.Name, v.Namespace, diskCount+numberOfAdditionalDisks, diskCountOutput)
+					}
+					return diskCountOutput, false, nil
+				}
+				d, err := task.DoRetryWithTimeout(t, bootTime, 30*time.Second)
+				log.FailOnError(err, "Failed to get number of disks in VM [%s] in namespace [%s] after retry", v.Name, v.Namespace)
+				diskCountAfterAddingDisks := d.(int)
+				log.InfoD("Number of disks in VM [%s] in namespace [%s] is [%d] after adding [%d] disks", v.Name, v.Namespace, diskCountAfterAddingDisks, numberOfAdditionalDisks)
+				dash.VerifyFatal(diskCountAfterAddingDisks, diskCount+numberOfAdditionalDisks, fmt.Sprintf("Verifying that VM [%s] in namespace [%s] has [%d] disks after restore", v.Name, v.Namespace, diskCount+numberOfAdditionalDisks))
+			}
+		})
+
+	})
+
+	JustAfterEach(func() {
+		log.Infof("Test execution completed")
+		defer EndPxBackupTorpedoTest(scheduledAppContexts)
+		ctx, err := backup.GetAdminCtxFromSecret()
+		log.FailOnError(err, "Fetching px-central-admin ctx")
+		opts := make(map[string]bool)
+		opts[SkipClusterScopedObjects] = true
+		RuleEnumerateReq := &api.RuleEnumerateRequest{
+			OrgId: BackupOrgID,
+		}
+		ruleList, err := Inst().Backup.EnumerateRule(ctx, RuleEnumerateReq)
+		for _, r := range ruleList.GetRules() {
+			log.Infof("Deleting rule [%s]", r.Name)
+			_, err := Inst().Backup.DeleteRule(ctx, &api.RuleDeleteRequest{
+				OrgId: BackupOrgID,
+				Name:  r.Name,
+				Uid:   r.Uid,
+			})
+			dash.VerifySafely(err, nil, fmt.Sprintf("Deleting rule [%s]", r.Name))
+		}
+		log.Info("Destroying scheduled apps on source cluster")
+		// TODO: Need to uncomment the below code once we have data validation for kubevirt VMs implemented
+		//err = DestroyAppsWithData(scheduledAppContexts, opts, controlChannel, errorGroup)
+		DestroyApps(scheduledAppContexts, opts)
+		CleanupCloudSettingsAndClusters(backupLocationMap, cloudCredName, cloudCredUID, ctx)
+	})
+})
+
+// CreatePVCsForVM creates PVCs for the VM
+func CreatePVCsForVM(vm kubevirtv1.VirtualMachine, numberOfPVCs int, storageClassName, resourceStorage string) ([]*corev1.PersistentVolumeClaim, error) {
```

Comment:
Can we move this to backup_helper or utilities?

Reply:
I thought about it, but this function maybe very specific to this test case and may not be a generic function

Reply:
In future it might be required to write helper functions for Kubevirt, if we keep it here it will end up in creating cyclic imports at that time.

Reply:
anyway, moved it to helper